nexus/src/test/java/com/example/nexus/service/AuthServiceImplTests.java

```diff
@@ -0,0 +1,214 @@
+package com.example.nexus.service;
+
+import com.example.nexus.constant.MessageConstants;
+import com.example.nexus.constant.RoleConstants;
+import com.example.nexus.exception.NotFoundException;
+import com.example.nexus.exception.UnauthorizedException;
+import com.example.nexus.exception.UserAlreadyExistsException;
+import com.example.nexus.mapper.RegisterMapper;
+import com.example.nexus.model.entity.Profile;
+import com.example.nexus.model.entity.Role;
+import com.example.nexus.model.entity.User;
+import com.example.nexus.model.payload.request.RegisterRequest;
+import com.example.nexus.repository.ProfileRepository;
+import com.example.nexus.repository.RoleRepository;
+import com.example.nexus.repository.UserRepository;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.ArgumentCaptor;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.springframework.security.crypto.password.PasswordEncoder;
+
+import java.util.ArrayList;
+import java.util.Optional;
+
+import static org.assertj.core.api.Assertions.*;
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.Mockito.*;
+
+@ExtendWith(MockitoExtension.class)
+public class AuthServiceImplTests {
+    private static String username;
+    private static String existingUsername;
+    private static String correctPassword;
+    private static String incorrectLengthPassword;
+    private static String noNumbersPassword;
+    private static String noUpperCaseLettersPassword;
+    private static String noLowerCaseLettersPassword;
+    private static String incorrectRepeatedPassword;
+    private static String firstName;
+    private static String lastName;
+    private static String passwordHash;
+    private static RegisterRequest registerRequest;
+    private static Profile profile;
```

Comment:
You can just set up only profile and just use it in tests. Dont nead  all of this different fields as static variables.